### PR TITLE
recreate existing control plane node correctly

### PR DESCRIPTION
**Pull request by sharifelgamal**

### CODE REVIEW

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -877,16 +877,26 @@ func validateRegistryMirror() {
...
+	var err error
+	if existing == nil {
+		cp := config.Node{
+			Port:              cc.KubernetesConfig.NodePort,
+			KubernetesVersion: getKubernetesVersion(&cc),
+			Name:              kubeNodeName,
+			ControlPlane:      true,
+			Worker:            true,
+		}
+		cc.Nodes = []config.Node{cp}
```
Comment (medyagh) [inline cmd/minikube/cmd/start.go:892]: can we return here, and then get rid of else for less code complexity?

Comment (sharifelgamal) [reply cmd/minikube/cmd/start.go:890]: yep, done

File: test/integration/aaa_download_only_test.go [hunk 2]
```diff
@@ -74,6 +74,7 @@ func TestDownloadOnly(t *testing.T) {
 
 					// skip for none, as none driver does not have preload feature.
 					if !NoneDriver() {
+						viper.Set("preload", true)
```
Comment (medyagh) [inline test/integration/aaa_download_only_test.go:77]: this is not the right way to do this. setting global flag. please pass this as an arg to PreloadExists. maybe as an optional argument
...preload bool

**Review by medyagh — CHANGES_REQUESTED** [review]

File: test/integration/aaa_download_only_test.go (hunk 2 above)
Comment (sharifelgamal) [reply test/integration/aaa_download_only_test.go:77]: is a hidden boolean trying to emulate a global flag really better than just using the flag?

**Comment by sharifelgamal** [comment]
Docker test is a known flake: https://github.com/kubernetes/minikube/issues/8014

**Pull request merged**